invocations/requirements.txt

```diff
@@ -1,7 +1,7 @@
 invoke~=1.7.1
 pip>=21.3
-poetry~=1.2.0
+poetry~=1.4.0
```

Comment:
@blakewaud I'm jumping to the latest Poetry since I noticed some deprecated changes and section we were using in _pyproject.toml_.

If this jump works well in main & meta, I suggest doing the move with all the other packages. (including using the newer `[build-system]` and stripping of `setuptools` dep)